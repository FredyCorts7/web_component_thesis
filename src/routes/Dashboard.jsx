import React from 'react';
import clsx from 'clsx';
import {
  makeStyles,
  CssBaseline,
  Drawer,
  AppBar,
  Toolbar,
  List,
  Typography,
  Divider,
  IconButton,
  Container,
} from '@material-ui/core';
import { MenuRounded, ChevronLeft } from '@material-ui/icons';
import { Redirect, Route, Switch, useRouteMatch } from 'react-router-dom';
import { useSelector } from 'react-redux';

import { ListItemNavigation } from '../components/ListItemNavigation';
import { InscriptionCreate } from '../containers/InscriptionCreate';
import { HomeDash } from '../containers/HomeDash';
import { InscriptionRead } from '../containers/InscriptionRead';
import { InscriptionReadMe } from '../containers/InscriptionReadMe';
import { PlanningComponents } from '../containers/PlanningComponents';
import PlanningMenus from '../containers/PlanningMenus';

const drawerWidth = 250;

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
  },
  toolbar: {
    paddingRight: 24, // keep right padding when drawer closed
  },
  toolbarIcon: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'flex-end',
    padding: '0 8px',
    ...theme.mixins.toolbar,
  },
  appBar: {
    backgroundColor: '#3c3c3c',
    zIndex: theme.zIndex.drawer + 1,
    transition: theme.transitions.create(['width', 'margin'], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
  },
  appBarShift: {
    marginLeft: drawerWidth,
    width: `calc(100% - ${drawerWidth}px)`,
    transition: theme.transitions.create(['width', 'margin'], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.enteringScreen,
    }),
  },
  menuButton: {
    marginRight: 36,
  },
  menuButtonHidden: {
    display: 'none',
  },
  title: {
    flexGrow: 1,
  },
  drawerPaper: {
    position: 'relative',
    whiteSpace: 'nowrap',
    width: drawerWidth,
    transition: theme.transitions.create('width', {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.enteringScreen,
    }),
  },
  drawerPaperClose: {
    overflowX: 'hidden',
    transition: theme.transitions.create('width', {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
    width: theme.spacing(7),
    [theme.breakpoints.up('sm')]: {
      width: theme.spacing(9),
    },
  },
  appBarSpacer: theme.mixins.toolbar,
  content: {
    flexGrow: 1,
    height: '100vh',
    overflow: 'auto',
  },
  container: {
    paddingTop: theme.spacing(4),
    paddingBottom: theme.spacing(4),
  },
  paper: {
    padding: theme.spacing(2),
    display: 'flex',
    overflow: 'auto',
    flexDirection: 'column',
  },
  fixedHeight: {
    height: 240,
  },
}));

export function Dashboard() {
  const classes = useStyles();

  const { titleNavbar } = useSelector((state) => state.uiReducer);

  const { path } = useRouteMatch();
  const [open, setOpen] = React.useState(false);
  const handleDrawerOpen = () => {
    setOpen(true);
  };
  const handleDrawerClose = () => {
    setOpen(false);
  };

  return (
    <div className={classes.root}>
      <CssBaseline />
      <AppBar
        position='absolute'
        className={clsx(classes.appBar, open && classes.appBarShift)}
      >
        <Toolbar className={classes.toolbar}>
          <IconButton
            edge='start'
            color='inherit'
            aria-label='open drawer'
            onClick={handleDrawerOpen}
            className={clsx(
              classes.menuButton,
              open && classes.menuButtonHidden
            )}
          >
            <MenuRounded />
          </IconButton>
          <Typography
            component='h1'
            variant='h6'
            color='inherit'
            noWrap
            className={classes.title}
          >
            {titleNavbar}
          </Typography>
        </Toolbar>
      </AppBar>

      <Drawer
        variant='permanent'
        classes={{
          paper: clsx(classes.drawerPaper, !open && classes.drawerPaperClose),
        }}
        open={open}
      >
        <div className={classes.toolbarIcon}>
          <IconButton onClick={handleDrawerClose}>
            <ChevronLeft />
          </IconButton>
        </div>
        <Divider />
        <List>
          <ListItemNavigation />
        </List>
      </Drawer>

      <main className={classes.content}>
        <div className={classes.appBarSpacer} />
        <Container maxWidth='lg' className={classes.container}>
          <Switch>
            <Route exact path={`${path}/home`} component={HomeDash} />

            <Route
              exact
              path={`${path}/inscription/create`}
              component={InscriptionCreate}
            />

            <Route
              exact
              path={`${path}/inscription/readme`}
              component={InscriptionReadMe}
            />

            <Route
              exact
              path={`${path}/inscription/read`}
              component={InscriptionRead}
            />

            <Route
              exact
              path={`${path}/planning/component`}
              component={PlanningComponents}
            />

            <Route
              exact
              path={`${path}/planning/menu`}
              component={PlanningMenus}
            />

            <Redirect to={`${path}/home`} />
          </Switch>
        </Container>
      </main>
    </div>
  );
}
