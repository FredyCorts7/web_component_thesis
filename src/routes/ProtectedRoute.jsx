import React from 'react';
import PropTypes from 'prop-types';
import { Redirect, Route } from 'react-router-dom';

export const ProtectedRoute = ({
  isAuthenticated,
  redirect,
  component: Component,
  ...rest
}) => {
  return (
    <Route
      {...rest}
      component={(props) =>
        isAuthenticated ? <Component {...props} /> : <Redirect to={redirect} />
      }
    />
  );
};

ProtectedRoute.propTypes = {
  isAuthenticated: PropTypes.bool.isRequired,
  redirect: PropTypes.string.isRequired,
  component: PropTypes.func.isRequired,
};
