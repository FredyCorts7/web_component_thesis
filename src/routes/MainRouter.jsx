import React from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Redirect,
} from 'react-router-dom';

import Dashboard from './Dashboard';

export const MainRouter = () => {
  return (
    <Router>
      <div>
        <Switch>
          {/* <Route exact path='/auth/login' component={} /> */}
          <Route path='/dashboard' component={Dashboard} />

          <Redirect to='/dashboard' />
        </Switch>
      </div>
    </Router>
  );
};
