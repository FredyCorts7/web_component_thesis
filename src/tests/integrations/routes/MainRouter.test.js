import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';

import configureStock from 'redux-mock-store';
import thunk from 'redux-thunk';
import '@testing-library/jest-dom';

import { MainRouter } from '../../../routes/MainRouter';
import {
  startSilentAuthentication,
  startDictionaryRead,
} from '../../../actions/authentication.action';

const middlewares = [thunk];
const mockStore = configureStock(middlewares);

const dictionary = {
  alphabet: [{ original: 'A', substitute: '2' }],
  numbers: [{ original: '1', substitute: '9' }],
};

jest.mock('../../../actions/authentication.action', () => ({
  startSilentAuthentication: jest.fn(),
  startDictionaryRead: jest.fn(),
}));

describe('Prueba de integración sobre el enrutador principal', () => {
  test('CP - Debería ...', () => {
    const initState = {
      authenticationReducer: {
        isAuthenticated: false,
        dictionary,
      },
      uiReducer: {
        showBackdrop: false,
      },
    };
    const store = mockStore(initState);
    store.dispatch = jest.fn();

    const wrapper = mount(
      <Provider store={store}>
        <MainRouter />
      </Provider>
    );

    expect(wrapper).toMatchSnapshot();
    expect(startSilentAuthentication).toHaveBeenCalledTimes(1);
  });
});
