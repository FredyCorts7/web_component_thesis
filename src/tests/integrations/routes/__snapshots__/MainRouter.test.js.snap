// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de integración sobre el enrutador principal CP18 - Debería actuar el <MainRouter/> según el estado global 1`] = `
<div>
  <main
    className="MuiGrid-root makeStyles-root-1 MuiGrid-container"
  >
    <div
      aria-hidden={true}
      className="MuiBackdrop-root makeStyles-backdrop-2"
      style={
        Object {
          "opacity": 0,
          "visibility": "hidden",
        }
      }
    >
      <div
        className="MuiCircularProgress-root MuiCircularProgress-indeterminate"
        role="progressbar"
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      >
        <svg
          className="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx={44}
            cy={44}
            fill="none"
            r={20.2}
            strokeWidth={3.6}
            style={Object {}}
          />
        </svg>
      </div>
    </div>
    <div
      className="MuiPaper-root MuiGrid-root makeStyles-backDark-3 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-7 MuiPaper-elevation1"
    >
      <div
        className="makeStyles-paper-4"
      >
        <div
          className="MuiBox-root MuiBox-root-10"
        >
          <div
            className="MuiChip-root makeStyles-root-11 MuiChip-outlined"
            onKeyDown={[Function]}
            onKeyUp={[Function]}
          >
            <div
              className="MuiAvatar-root MuiAvatar-circle MuiChip-avatar MuiAvatar-colorDefault"
            >
              2
            </div>
            <span
              className="MuiChip-label"
            >
              A
            </span>
          </div>
        </div>
        <div
          className="MuiBox-root MuiBox-root-12"
        >
          <div
            className="MuiChip-root makeStyles-root-11 MuiChip-outlined"
            onKeyDown={[Function]}
            onKeyUp={[Function]}
          >
            <div
              className="MuiAvatar-root MuiAvatar-circle MuiChip-avatar MuiAvatar-colorDefault"
            >
              9
            </div>
            <span
              className="MuiChip-label"
            >
              1
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiPaper-root MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-5 MuiPaper-elevation1"
    >
      <div
        className="makeStyles-paper-4"
      >
        <h1
          className="MuiTypography-root MuiTypography-h5"
        >
          Inicio de Sesión
        </h1>
        <form
          className="makeStyles-form-5"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
            data-cy="email"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink={false}
            >
              Correo
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="email"
                autoFocus={true}
                className="MuiInputBase-input MuiOutlinedInput-input"
                disabled={false}
                name="email"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-13 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-15"
                >
                  <span>
                    Correo
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink={false}
              htmlFor="outlined-adornment-password"
            >
              Contraseña
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
              data-cy="password"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                disabled={false}
                id="outlined-adornment-password"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="password"
                value=""
              />
              <div
                className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
              >
                <button
                  aria-label="toggle password visibility"
                  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 6.5c2.76 0 5 2.24 5 5 0 .51-.1 1-.24 1.46l3.06 3.06c1.39-1.23 2.49-2.77 3.18-4.53C21.27 7.11 17 4 12 4c-1.27 0-2.49.2-3.64.57l2.17 2.17c.47-.14.96-.24 1.47-.24zM2.71 3.16c-.39.39-.39 1.02 0 1.41l1.97 1.97C3.06 7.83 1.77 9.53 1 11.5 2.73 15.89 7 19 12 19c1.52 0 2.97-.3 4.31-.82l2.72 2.72c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L4.13 3.16c-.39-.39-1.03-.39-1.42 0zM12 16.5c-2.76 0-5-2.24-5-5 0-.77.18-1.5.49-2.14l1.57 1.57c-.03.18-.06.37-.06.57 0 1.66 1.34 3 3 3 .2 0 .38-.03.57-.07L14.14 16c-.65.32-1.37.5-2.14.5zm2.97-5.33c-.15-1.4-1.25-2.49-2.64-2.64l2.64 2.64z"
                      />
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-13 MuiOutlinedInput-notchedOutline"
                style={
                  Object {
                    "paddingLeft": 8,
                  }
                }
              >
                <legend
                  className="PrivateNotchedOutline-legend-14"
                  style={
                    Object {
                      "width": 0.01,
                    }
                  }
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-9 MuiButton-fullWidth"
            data-cy="btn-login"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              <span
                className="MuiButton-startIcon MuiButton-iconSizeMedium"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM9.29 16.29L5.7 12.7a.9959.9959 0 010-1.41c.39-.39 1.02-.39 1.41 0L10 14.17l6.88-6.88c.39-.39 1.02-.39 1.41 0 .39.39.39 1.02 0 1.41l-7.59 7.59c-.38.39-1.02.39-1.41 0z"
                  />
                </svg>
              </span>
              Iniciar Sesión
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </form>
      </div>
    </div>
  </main>
</div>
`;
